fs/fs_test.go

```diff
@@ -0,0 +1,100 @@
+
+func TestFS(t *testing.T) {
+	expectedOutput := longFileContent + longFileContent + "\n"
+
+	Convey("Given two files each containing a line over 65536 characters long, they can be concatenated and compressed.", t, func() {
+		inputs, output, outputPath := buildTestFiles(t)
+
+		err := combine.ConcatenateAndCompress(inputs, output)
+		So(err, ShouldBeNil)
+
+		_, err = os.Stat(outputPath)
+		So(err, ShouldBeNil)
+
+		Convey("With a default buffer length, the content cannot be read.", func() {
```

Comment:
In TDD you don't keep the old test and old code. You write a test for fs.ReadCompressedFile with a too-long-line, and note (in your head) that it fails. Then you fix the fs.ReadCompressedFile implementation and note  (in your head) the test now passes.

So get rid of readCompressedFileWithArbitraryBuffer(), and actually test fs.ReadCompressedFile.

Also, the fix being a buffer that's 1000000 bytes long is too unexplained and arbitrary. When declaring the size constant, give it some documentation and base the number on some justification. Eg. "the max path length is 4096 characters long, and we at least have to deal with files that contain 2 columns of paths, so to be safe we make this 3 * 4096". (Though I note you're testing with 65536 long lines; come up with an alternative justification for it being even bigger than that if necessary.)